src/HealthGPS/energy_balance_model.cpp

```diff
@@ -35,66 +46,163 @@ void EnergyBalanceModel::generate_risk_factors([[maybe_unused]] RuntimeContext &
+    double H = H_0;
+
+    // TODO: Update physical activity level.
+    double PAL = PAL_0;
+
+    // Compute nutrient intakes from food intakes.
+    std::unordered_map<core::Identifier, double> nutrient_intakes;
+    for (const auto &coefficient : energy_equation_) {
+        nutrient_intakes[coefficient.first] = 0.0;
+    }
+    for (const auto &equation : nutrient_equations_) {
+        double food_intake = person.get_risk_factor_value(equation.first);
 
-        // Compute energy intake from nutrient intakes.
-        for (const auto &coefficient : energy_equation_) {
-            double delta_energy = nutrient_intakes[coefficient.first] * coefficient.second;
-            energy_intake += delta_energy;
+        for (const auto &coefficient : equation.second) {
+            double delta_nutrient = food_intake * coefficient.second;
+            nutrient_intakes.at(coefficient.first) += delta_nutrient;
         }
-
-        // TODO: model after energy intake stage
     }
-}
 
-std::map<core::Identifier, double>
-EnergyBalanceModel::get_current_risk_factors(const HierarchicalMapping &mapping, Person &entity,
-                                             int time_year) const {
-    auto entity_risk_factors = std::map<core::Identifier, double>();
-    entity_risk_factors.emplace(InterceptKey, entity.get_risk_factor_value(InterceptKey));
-    for (const auto &factor : mapping) {
-        if (factor.is_dynamic_factor()) {
-            entity_risk_factors.emplace(factor.key(), time_year - 1);
-        } else {
-            entity_risk_factors.emplace(factor.key(), entity.get_risk_factor_value(factor.key()));
-        }
+    // Update energy intake and carbohydrate intake.
+    double EI = 0.0;
+    double CI = nutrient_intakes.at(CI_key);
+    for (const auto &coefficient : energy_equation_) {
+        double delta_energy = nutrient_intakes.at(coefficient.first) * coefficient.second;
+        EI += delta_energy;
     }
 
-    return entity_risk_factors;
+    // Update glycogen and water.
+    double k_G = CI_0 / (G_0 * G_0);
+    double G = sqrt(CI / k_G);
+    double W = 2.7 * G;
+
+    // Update extracellular fluid.
+    double Na_b = 4000.0;
+    double Na_f = Na_b * EI / EI_0;
+    double Delta_Na_diet = Na_f - Na_b;
+    double ECF = ECF_0 + (Delta_Na_diet - xi_CI * (1.0 - CI / CI_0)) / xi_Na;
+
+    // Update resting metabolic rate (Mifflin-St Jeor).
+    double RMR = 9.99 * BW_0 + 6.25 * H * 100.0 - 4.92 * person.age;
+    RMR += person.gender == core::Gender::male ? 5.0 : -161.0;
+    RMR *= 4.184; // kcal to kJ
+
+    double delta_0 = ((1.0 - beta_TEF) * PAL - 1.0) * RMR / BW_0;
+
+    // Update thermic effect of food and adaptive thermogenesis.
+    double TEF = beta_TEF * EI - EI_0;
+    double AT = beta_AT * EI - EI_0;
+
+    // TODO: Update intercept value.
+    const double K = 0.0; // TODO: Intercept value.
+
+    // Energy partitioning.
+    const double C = 10.4 * rho_L / rho_F;
+    const double p = C / (C + F_0);
```

Comment:
I think all the equations and update steps should be made into separate functions for testability.